src/Ledger/Chain/Properties.agda

```diff
@@ -8,24 +8,51 @@ module Ledger.Chain.Properties
   (txs : _) (open TransactionStructure txs)
   (abs : AbstractFunctions txs) (open AbstractFunctions abs)
   where
-
+open import Ledger.Properties txs abs using (getLState)
+open import Ledger.Enact govStructure
 open import Ledger.Chain txs abs
 open import Ledger.Epoch txs abs
 open import Ledger.Epoch.Properties txs abs
+open import Ledger.Ledger txs abs
 open import Ledger.Ledger.Properties txs abs
 
 open Computational ⦃...⦄
 
-module _ {nes : NewEpochState} {e : Epoch} where
-
 instance
   Computational-CHAIN : Computational _⊢_⇀⦇_,CHAIN⦈_ String
-  Computational-CHAIN .computeProof Γ s b = do
-    _ , neStep ← map₁ ⊥-elim $ computeProof {STS = _⊢_⇀⦇_,NEWEPOCH⦈_} _ _ _
-    _ , lsStep ← computeProof _ _ _
-    success (_ , CHAIN neStep lsStep)
-  Computational-CHAIN .completeness Γ s b s' (CHAIN neStep lsStep)
-    with recomputeProof neStep | completeness _ _ _ _ neStep
-  ... | _         | refl
-    with recomputeProof lsStep | completeness _ _ _ _ lsStep
-  ... | success _ | refl = refl
+  Computational-CHAIN .computeProof Γ
+    s@record { newEpochState = nes }
+    b@record { ts = ts ; slot = slot } = do
+         nes' , neStep ← map₁ ⊥-elim $ computeProof {STS = _⊢_⇀⦇_,NEWEPOCH⦈_} _ _ _
+         ls' , lsStep ← computeProof _ (getLState nes') ts
+         case ((totalRefScriptsSize (EpochState.ls (NewEpochState.epochState nes')) ts)
+                ≤? (PParams.maxRefScriptSizePerBlock (proj₁ (EnactState.pparams
+                     (EpochState.es (NewEpochState.epochState nes')))))) of λ where
+           (no ¬p) → failure "totalRefScriptsSize > maxRefScriptSizePerBlock"
+           (yes p) → success (_ , CHAIN p neStep lsStep)
+
+  Computational-CHAIN .completeness .tt s b
+    .(record { newEpochState = record { lastEpoch = NewEpochState.lastEpoch nes
+                                      ; epochState = record  { acnt = EpochState.acnt (NewEpochState.epochState nes)
+                                                             ; ss = EpochState.ss (NewEpochState.epochState nes)
+                                                             ; ls = _
+                                                             ; es = EpochState.es (NewEpochState.epochState nes)
+                                                             ; fut = EpochState.fut (NewEpochState.epochState nes)
+                                                             }
+                                      ; ru = NewEpochState.ru nes
+                                      }
+             }) (CHAIN {nes = nes} p neStep lsStep) = goal
+    where
+    open NewEpochState nes
+    open EpochState epochState
+    goal : map proj₁ (Computational.computeProof Computational-CHAIN tt s b)
+           ≡ success record { newEpochState = ⟦ lastEpoch , ⟦ acnt , ss , _ , EpochState.es epochState , fut ⟧ᵉ' , ru ⟧ⁿᵉ }
```

Comment:
```suggestion
           ≡ success record { newEpochState = _ }
```

No need to repeat this. In fact, even the entire `record` expression can be turned into an underscore.